packages/types/src/ui/common.ts

```diff
@@ -0,0 +1,3 @@
+export type UIEvent = Event & {
+  currentTarget: EventTarget & HTMLInputElement
+} & { key?: string } & { target?: any }
```

Comment:
Haha, finally snapped huh?

Reply:
I think you might be better served here by something like:

```typescript
type BudibaseEventHandler<E extends Event = Event, T extends EventTarget = Element, V extends EventTarget = Element> = (
  event: E & { currentTarget: EventTarget & T, target: EventTarget & V }
)
```

To follow Svelte's own types:

![CleanShot 2025-02-06 at 17 10 01@2x](https://github.com/user-attachments/assets/051e4587-6932-436d-a3bb-65aa62994e87)